components/post_markdown/system_message_helpers.tsx

```diff
@@ -376,11 +380,11 @@ const systemMessageRenderers = {
     [Posts.POST_TYPES.ME]: renderMeMessage,
 };
 
-export function renderSystemMessage(post, channel, isUserCanManageMembers) {
+export function renderSystemMessage(post: Post, channel?: Channel, isUserCanManageMembers?: boolean): ReactNode {
```

Comment:
Can this ever actually be `undefined`?

Reply:
since the channel is the optional props in the `PostMarkdown` we need to either make this `channel` optional or make the `channel` props in the component required.

Reply:
We can make `channel` in the component props required, as it was originally always passed in by redux. I think we might need to make `channelId` required as well for `OwnProps` in the connector.

Reply:
@devinbinnie I have tried this but all the tests have type errors. since the channel we assume we have is handled in the `index` container and the tests imported the component itself. even `TestHelper.getPostMock` doesn't handle `channelId` properly.

Reply:
Since in `OwnProps` we are assuming that `channelId` is defined, I'm not sure why we can't assume it's defined in the component as well. If it's just the tests that are breaking, can we make sure that we always pass some default `channelId` in every test? Doesn't matter what the value is if the test doesn't care.